pkg/client/connector/userd_grpc/service.go

```diff
@@ -171,26 +171,18 @@ func (s *service) Logout(ctx context.Context, _ *empty.Empty) (*empty.Empty, err
 	return &empty.Empty{}, nil
 }
 
-func (s *service) getCloudAccessToken(ctx context.Context, autoLogin bool) (string, error) {
-	token, err := s.sharedState.LoginExecutor.GetToken(ctx)
-	if autoLogin && err != nil {
+func (s *service) GetCloudUserInfo(ctx context.Context, req *rpc.UserInfoRequest) (*rpc.UserInfo, error) {
+	ctx = s.callCtx(ctx, "GetCloudUserInfo")
+	info, err := s.sharedState.LoginExecutor.GetUserInfo(ctx, req.GetRefresh())
+	if req.GetAutoLogin() && err != nil {
 		if _err := s.sharedState.LoginExecutor.Login(ctx); _err == nil {
-			token, err = s.sharedState.LoginExecutor.GetToken(ctx)
+			info, err = s.sharedState.LoginExecutor.GetUserInfo(ctx, req.GetRefresh())
```

Comment:
This means that all kinds of errors that the `LoginExecutor.Login()` can return are lost. Is that intentional?

Reply:
Indeed it is.